components/form/CustomerForm.tsx

```diff
@@ -18,80 +27,110 @@ const validateCustomerFields = (fields: CustomerFields): Customer => {
     throw Error("Invalid customer form: missing required fields")
 }
 
-export function CreateCustomerForm({
-    afterSubmit,
+function CustomerForm({
+    onSubmit,
     onCancel,
-}: CreateCustomerFormProps): JSX.Element {
-    const [customerFields, setCustomerFields] = useState<CustomerFields>({})
+    customer: customerOrNull,
+}: CustomerFormProps) {
+    const [customerFields, setCustomerFields] = useState<CustomerFields>(
+        customerOrNull || {}
+    )
+
+    const [errorMessage, setErrorMessage] = useState<string>("")
+    const errorHandler = (error: Error) => setErrorMessage(`${error}`)
+
+    const abortSubmission = onCancel && onCancel
+
+    return (
+        <>
```

Comment:
The edit page might look a bit nicer if, instead of an empty fragment `<></>` the form was wrapped in the same kind of `Flex` that we have in `TimesheetForm.tsx`:

```typescript
<Flex
    flexDirection="column"
    backgroundColor="white"
    border="1px solid"
    borderColor="gray.400"
    borderRadius="0.2rem"
    padding="1rem 1rem"
>
```